phys/module_sf_noahdrv.F

```diff
@@ -977,6 +977,7 @@ SUBROUTINE lsm(DZ8W,QV3D,P8W3D,T3D,TSK,                      &
 #endif
 
           IF (rdlai2d) THEN
+             if(lai(i,j).eq.0.0) lai(i,j) = 0.01 
```

Comment:
This is essentially a consistency check on input datasets and could be made more general:

`if(lai(i,j).eq.0.0) lai(i,j) = 0.01`

replaced with

`
if(shdfac == 0.0) lai(i,j) = 0.0
if(shdfac > 0.0 .and. lai(i,j) == 0.0) lai(i,j) = 0.01
`